include/hal/library/requester/reqasymsignlib.h

```diff
@@ -10,7 +10,8 @@
 #include "hal/base.h"
 #include "internal/libspdm_lib_config.h"
 
-#if LIBSPDM_ENABLE_CAPABILITY_MUT_AUTH_CAP
+#if (LIBSPDM_ENABLE_CAPABILITY_ENCAP_CAP) && \
+    ((LIBSPDM_ENABLE_CAPABILITY_MUT_AUTH_CAP) || (LIBSPDM_ENABLE_CAPABILITY_ENDPOINT_INFO_CAP))
```

Comment:
Supporting ENDPOINT_INFO does not mean to support signing.
The ENDPOINT_INFO could be unsigned.